addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -334,63 +340,64 @@ def unpickable_item(self, reason, product_id=None, location_id=None,
         picking = move_lines.mapped('picking_id')
         picking.message_post(body=msg)
 
-        if picking_type_id is None:
-            picking_type_id = ResUsers.get_user_warehouse().int_type_id.id
-
-        if picking.batch_id != self:
-            raise ValidationError(_('Move line is not part of the batch.'))
-
-        if picking.state in ['cancel', 'done']:
-            raise ValidationError(_('Cannot mark a move line as unpickable '
-                                    'when it is part of a completed Picking.'))
-
-        original_picking_id = None
-
-        if len(picking.move_line_ids - move_lines):
-            # Create a backorder for the affected move lines if
-            # there are move lines that are not affected
-            original_picking_id = picking.id
-            picking = picking._backorder_movelines(move_lines)
-
-        # By default the pick is unreserved
-        picking._refine_picking(reason)
-
-        group = Group.get_group(group_identifier=reason,
-                                create=True)
-
-        # create new "investigation pick"
-        Picking.with_context(allow_partial=allow_partial)\
-               .create_picking(quant_ids=quants.mapped('id'),
-                               location_id=location.id,
-                               picking_type_id=picking_type_id,
-                               group_id=group.id)
-
-        # Try to re-assing the picking after, by creating the
-        # investigation, we've reserved the problematic stock
-        picking.action_assign()
-
-        if picking.state != 'assigned':
-            # Remove the picking from the batch as it cannot be
-            # processed for lack of stock; we do so to be able
-            # to terminate the batch and let the user create a
-            # new batch for himself
-            picking.batch_id = False
-        elif original_picking_id is not None:
-            # A backorder has been created, but the stock is
-            # available; get rid of the backorder after linking the
-            # move lines to the original picking, so it can be
-            # directly processed
-            picking.move_line_ids.write({'picking_id': original_picking_id})
-            picking.move_lines.write({'picking_id': original_picking_id})
-            picking.unlink()
-
-        # If the batch does not contain any remaining picking to do,
-        # it can be set as done
-        remaining_pickings = self.picking_ids.filtered(
-            lambda x: x.state in ['assigned'])
-
-        if not remaining_pickings.exists():
-            self.state = 'done'
+        if raise_stock_investigation:
```

Comment:
Can you just `return True` otherwise instead of having a big clause, for the sake of readability?